earthfile2llb/converter.go

```diff
@@ -779,7 +779,7 @@ func (c *Converter) canSave(ctx context.Context, saveAsLocalTo string) (bool, er
 	if !basePathExists {
 		return false, fmt.Errorf("no such directory: %s", basepath)
 	}
-	basepath += "/"
+	basepath += string(filepath.Separator)
```

Comment:
This fixes a bug with `SAVE ARTIFACT` failing to write a specified path on Windows. I tried a simple example, on the current `main` version of earthly, I get an error like this on Powershell:
```
Error: build target: build main: bkClient.Build: failed to solve: Earthfile line 22:4 apply SAVE ARTIFACT: unable to save to ./hello/hi; path must be located under .
in              +save
```
But on this branch with the file separator fix, I see the output looks appropriate for Windows now:
```
Artifact +save/hello output as .\hello\hi
```

Reply:
Note that this still uses forward-slash in the Earthfile, here's the one from my test:
```earthly
VERSION 0.6

save:
    FROM alpine
    RUN echo "hi" > hello
    SAVE ARTIFACT hello AS LOCAL ./hello/hi
```

Reply:
We have SAVE ARTIFACT ... AS LOCAL /with/path/like/that using forward slashes and this PR handles them both under Windows.
Or in other words, keeping forward slashes inside Earthfile are interpreted correctly as file path when running under Windows with the fixes of this PR 👍